datafusion/expr/src/expr.rs

```diff
@@ -1692,6 +1692,24 @@ fn create_names(exprs: &[Expr]) -> Result<String> {
         .join(", "))
 }
 
+/// Whether the given expression is volatile, i.e. whether it can return different results
+/// when evaluated multiple times with the same input.
+#[allow(clippy::match_like_matches_macro)]
+pub fn is_volatile(expr: &Expr) -> bool {
+    match expr {
+        Expr::ScalarFunction(func) => match func.func_def {
+            ScalarFunctionDefinition::BuiltIn(func) => match func {
```

Comment:
I suggest moving this logic into  a method like `ScalarFunctionDefinition::volatility` directly ?

https://github.com/apache/arrow-datafusion/blob/7f312c8a1d82b19f03c640e4a5d7d6437b2ee835/datafusion/expr/src/expr.rs#L339-L348

Reply:
Okay.